app/questionnaire/navigation.py

```diff
@@ -141,7 +141,7 @@ def _generate_repeated_items(self, link_names, first_location, completed_id, no_
             location.group_instance = i
             is_current_group_instance = is_current_group and i == current_group_instance
 
-            if link_names:
+            if link_names and link_names.get(i):
```

Comment:
Problem is although this would hide it in the navigation on the right, you would still get it if you went through the survey.